scripts/setup-ilab-macos.sh

```diff
@@ -0,0 +1,156 @@
+        echo "Error: This script is designed to run on macOS only."
+        return 1
+    fi
+
+    # Try checking python3.11 first (priority), then python3.10, then python3
+    if check_python_version python3.11; then
+        PYTHON_BIN="python3.11"
+    elif check_python_version python3.10; then
+        PYTHON_BIN="python3.10"
+    elif check_python_version python3; then
+        PYTHON_BIN="python3"
+    else
+        echo "Error: Python 3.10 or 3.11 is required but not found."
+        return 1
+    fi
+
+    echo "Using Python binary: $PYTHON_BIN"
+    echo "Detected CPU brand: $CPU_BRAND"
+    echo "Detected architecture: $OS_ARCH"
+
+    # Create installation directory if it doesn't exist
+    if [ ! -d "$INSTALL_DIR" ]; then
+        echo "Creating installation directory: $INSTALL_DIR"
+        mkdir -p "$INSTALL_DIR"
+    fi
+
+    # Switch to installation directory
+    cd "$INSTALL_DIR" || { echo "Error: Failed to change directory to $INSTALL_DIR"; return 1; }
+
+    # Create and activate virtual environment using the correct Python version
+    echo "Creating and activating virtual environment: venv"
+    $PYTHON_BIN -m venv --upgrade-deps "venv"
+    if [ -f "$INSTALL_DIR/venv/bin/activate" ]; then
+    # shellcheck source=/dev/null
+        source "$INSTALL_DIR/venv/bin/activate" || return 1
+    else
+        echo "Error: $INSTALL_DIR/venv/bin/activate not found."
+        return 1
+    fi
+
+    #  Apple Silicon (M1/M2/M3)
+    if [[ "$OS_ARCH" = "arm64" ]] && [[ "$CPU_BRAND" == *"Apple"* ]]; then
+        echo "Apple Silicon Mac detected. Checking the environment setup correctly..."
+
+        # Check that Python binary supports arm64
+        check_python_macho
+
+       # Verify that both Python and terminal are running as ARM64
+        PYTHON_ARCH=$($PYTHON_BIN -c 'import platform; print(platform.machine())')
+        if [ "$PYTHON_ARCH" != "arm64" ] || [ "$(arch)" != "arm64" ]; then
+            echo "Error: Either Python or terminal is not running in ARM64 mode."
+            echo "Please ensure both your Python environment and terminal are correctly set up for Apple Silicon."
+            echo "You may need to configure env or reinstall Python as an ARM64 binary, and ensure the terminal is running natively on Apple Silicon."
+            deactivate
+            return 1
+        fi
+
+        echo "System environment is correctly configured for Apple Silicon. Installing InstructLab with Metal support..."
+
+        # Clean llama_cpp_python cache and install with MPS support
+        pip cache remove llama_cpp_python
+        pip install 'instructlab[mps]'
+        echo "InstructLab installed with Apple Metal support."
+
+    elif [[ "$OS_ARCH" = "x86_64" ]] && [[ "$CPU_BRAND" == *"Intel"* ]]; then
+        # Intel Mac
+        echo "Intel Mac detected. Installing InstructLab with Intel CPU..."
+        CMAKE_ARGS="-DLLAMA_METAL=off" pip install instructlab
+        echo "InstructLab installed with Intel CPU support."
```

Comment:
we don't technically support Intel Mac, so I am wary to put this into our codebase